pkgs/by-name/fs/fsnotifier/package.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
+  version = "2024.2.0";
+  pname = "fsnotifier";
+
+  src = fetchFromGitHub {
+    owner = "JetBrains";
+    repo = "intellij-community";
+    rev = "0f6d9ccb67b8fcad0d802cd76209d503c4ed66a6";
+    sha256 = "3TAiVvKi50JQRrVG6J7LUJKTiuOTDyKt4DhoA1QmbrM=";
+    sparseCheckout = [ "native/fsNotifier/linux" ];
+  };
+
+  # fix for hard-links in nix-store, https://github.com/JetBrains/intellij-community/pull/2171
+  patches = [ ./fsnotifier.patch ];
```

Comment:
This is a large patch to include without mentioning in the package description. Maybe it can be put behind an overridable flag?